### PR TITLE
NRPT-227, NRPT-230: Fixes dateAdded and addedBy clobbering on subsequ…

### PR DESCRIPTION
**Pull request by marklise**
…ent importer runs.  Added note and stub about setting a RefId for a mine.  Fixes runs where the API calls post without setting auth_payload correctly.

Couldn't split out the bugs as they are directly related to each other. auth_payload will be set on every request so no need for check-then-default to SYSTEM_USER. Added some documentation on a sample mine record, there might be a better spot for it.

### CODE REVIEW

File: api/src/integrations/core/datasource.js [hunk 1]
```diff
@@ -25,10 +25,12 @@ class CoreDataSource {
    * Creates an instance of CoreDataSource.
    *
    * @param {*} taskAuditRecord audit record hook for this import instance
+   * @param {*} auth_payload information about the user account that started this update.
    * @memberof CoreDataSource
    */
-  constructor(taskAuditRecord) {
+  constructor(taskAuditRecord, auth_payload) {
```
Comment (KitArmstrong) [inline api/src/integrations/core/datasource.js:31]: On the other objects the auth_payload is the param. Might be nice to make them align.

File: api/src/integrations/core/mine-utils.js [hunk 2]
```diff
@@ -58,6 +59,8 @@ class Mines extends BaseRecordUtils {
       location : { type: 'Point', coordinates: mineRecord.coordinates },
       permittee: this.getParty(MINE_PARTY_PERMITTEE, mineRecord, permit),
       type: '',
+      // TODO:  Without this, subsequent runs of the mine importer will duplicate mines every time it's run
```
Comment (KitArmstrong) [inline api/src/integrations/core/mine-utils.js:62]: Does this still need to be done?

File: api/src/integrations/core/datasource.js [hunk 3]
```diff
@@ -58,6 +60,20 @@ class CoreDataSource {
 
   /**
    * Main function that runs all necessary operations to update Core records.
+   * Sample Record Data for reference
+      {
+        _schemaName: "Mine",
+        _sourceRefId: "abc123",
+        name: "Test Mine",
+        permitNumbers: ['M-209'],
```
Comment (KitArmstrong) [inline api/src/integrations/core/datasource.js:68]: it is now `permitNumber: string`.

File: api/src/integrations/core/mine-utils.js (hunk 2 above)
Comment (marklise) [reply api/src/integrations/core/mine-utils.js:62]: Yes

File: api/src/integrations/core/datasource.js (hunk 1 above)
Comment (marklise) [reply api/src/integrations/core/datasource.js:31]: I wanted to limit the amount of adjustments I made - cleanup ticket perhaps?

File: api/src/integrations/core/datasource.js (hunk 3 above)
Comment (marklise) [reply api/src/integrations/core/datasource.js]: Fixed in e40a974

File: api/src/integrations/core/mine-utils.js (hunk 2 above)
Comment (marklise) [reply api/src/integrations/core/mine-utils.js:62]: Ticket: https://bcmines.atlassian.net/browse/NRPT-232

**Pull request merged**